website/server/controllers/api-v4/news.js

```diff
@@ -0,0 +1,257 @@
+import {
+  NotFound,
+} from '../../libs/errors';
+
+const api = {};
+
+/**
+ * @apiDefine postIdRequired
+ * @apiError (400) {BadRequest} postIdRequired A postId is required
+ */
+
+/**
+ * @api {get} /api/v4/news Get latest Bailey announcement
+ * @apiName GetNews
+ * @apiGroup News
+ *
+ * @apiParam (Query) {Number} [page] This parameter can be used to specify the page number
+ *           (the initial page is number 0 and not required).
+ *
+ * @apiSuccess {Object} html Latest Bailey html
+ *
+ */
+api.getNews = {
+  method: 'GET',
+  url: '/news',
+  middlewares: [authWithHeaders({
+    optional: true,
+  })],
+  noLanguage: true,
+  async handler (req, res) {
+    const { user } = res.locals;
+    const { page } = req.query;
+
+    let isAdmin = false;
+    if (user && user.contributor) {
+      isAdmin = user.contributor.admin;
+    }
+    const results = await NewsPost.getNews(isAdmin, { page });
+    res.respond(200, results);
+  },
+};
+
+
+/**
+ * @api {post} /api/v4/news create a new news post
+ * @apiName CreateNewsPost
+ * @apiGroup News
+ *
+ * @apiSuccess {Object} data The create news post (See <a href="https://github.com/HabitRPG/habitica/blob/develop/website/server/models/newsPost.js" target="_blank">/website/server/models/newsPost.js</a>)
+ *
+ * @apiSuccessExample {json} Post:
+ *     HTTP/1.1 200 OK
+ *     {
+ *       "title": "News Title",
+ *       ...
+ *     }
+ *
+ * @apiPermission Admin
+ */
+api.createNews = {
+  method: 'POST',
+  url: '/news',
+  middlewares: [authWithHeaders(), ensureNewsPoster],
+  async handler (req, res) {
+    const postData = {
+      title: req.body.title,
+      publishDate: req.body.publishDate,
+      published: req.body.published,
+      credits: req.body.credits,
+      text: req.body.text,
+    };
+
+    const newsPost = new NewsPost(NewsPost.sanitize(postData));
+    await newsPost.save();
+
+    if (newsPost.published) {
+      NewsPost.updateLastNewsPost(newsPost);
+    }
+
+    res.respond(201, newsPost.toJSON());
+  },
+};
+
+/**
+ * @api {get} /api/v4/news/:postId get news post
+ * @apiName GetNewsPost
+ * @apiGroup News
+ *
+ * @apiParam (Path) {String} postId The posts _id
+ *
+ * @apiSuccess {Object} data The news post (See <a href="https://github.com/HabitRPG/habitica/blob/develop/website/server/models/newsPost.js" target="_blank">/website/server/models/newsPost.js</a>)
+ *
+ * @apiSuccessExample {json} Post:
+ *     HTTP/1.1 200 OK
+ *     {
+ *       "title": "News Title",
+ *       ...
+ *     }
+ *
+ * @apiUse postIdRequired
+ *
+ */
+api.getPost = {
+  method: 'GET',
+  url: '/news/:postId',
+  middlewares: [authWithHeaders({
+    optional: true,
+  })],
+  noLanguage: true,
+  async handler (req, res) {
+    req.checkParams('postId', apiError('postIdRequired')).notEmpty();
+    const { user } = res.locals;
+    let isAdmin = false;
+    if (user && user.contributor) {
+      isAdmin = user.contributor.admin;
```

Comment:
should this be `contributor.newsPoster`? Also this check is duplicated, I think we could create a function like the `user.isAdmin()` one in the user model

Reply:
Looks like there's one alreadt 🎉 , sorry i had missed it, then we could use that one